lasagne/layers/pool.py

```diff
@@ -127,6 +127,13 @@ def __init__(self, incoming, pool_size, stride=None, pad=0,
                  ignore_border=True, mode='max', **kwargs):
         super(Pool1DLayer, self).__init__(incoming, **kwargs)
 
+        if len(self.input_shape) != 3:
+                raise ValueError("Tried to create a 1D pooling layer with "
+                                "input shape %r. Expected 3 input dimensions "
+                                "(batchsize, channels, 1 spatial dimensions)." %
+                                (self.input_shape))
```

Comment:
This will throw a `TypeError: not all arguments converted during string formatting`. You need to write `(self.input_shape,)` so it's not mistaking the elements of `self.input_shape` to be the string formatting arguments.